controllers/operator_controller.go

```diff
@@ -121,7 +116,15 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 		})
 		return ctrl.Result{}, nil
 	}
-
+	if r.Resolver == nil {
```

Comment:
Keeping `Resolver` as a dependency for Operator reconciler, and passing in a nil value in main.go, since the tests pass in a fakeResolver with a `testEntitySource`. Ideally I'd like to use mocking for interfaces instead of passing in a nil value, but then kept it like this since this is the first pass implementation that just makes a new entitySource every time the reconciler runs, I.e we'll probably optimize the entire structure soon and it's probably worth waiting and seeing which way this molds before restricting the test suite to fit any mold.

Reply:
Playing devil's advocate: if your test suite is so dependent on the particular implementation details, it's too fragile and perhaps is testing at the wrong level of abstraction. A test that sets up inputs, runs a reconcile loop, and asserts over outputs should be robust to choices like how optimized the entity source is.

Reply:
> Playing devil's advocate: if your test suite is so dependent on the particular implementation details, it's too fragile and perhaps is testing at the wrong level of abstraction. A test that sets up inputs, runs a reconcile loop, and asserts over outputs should be robust to choices like how optimized the entity source is.

Generally yes I agree. Looks like when the entitySources was either [static](https://github.com/operator-framework/operator-controller/blob/e596002d68d6cbf8f1408ca06f04909ed8e0161d/main.go#L94), or had the [appearance of being static](https://github.com/operator-framework/operator-controller/blob/main/main.go#L108) since there was a controller updating the content of the entitySource, it made sense to structure the unit tests in a way that takes advantage of the [assumption that entitySource is going to be static](https://github.com/operator-framework/operator-controller/blob/a252719a953c59063947977f88d8bdbcde96887f/controllers/operator_controller_test.go#L36) from the reconciler's point of view. With [our wish to do away with that controller structure for creating entities](https://github.com/operator-framework/operator-controller/pull/141#issuecomment-1516407385) to get us started, it's also highlighting what @stevekuznetsov mentioned here. Hence the call out from me about passing in `nil` for that field in the main caller, because we'll probably have to rethinking the layout of the test suite (which I'd argue is out of scope for this PR but def should be addressed in a follow up PR)

Reply:
Going to point to a couple comments I made as a response to another comment:
- https://github.com/operator-framework/operator-controller/pull/182#discussion_r1177068140
- https://github.com/operator-framework/operator-controller/pull/182#discussion_r1177071919

The gist is that I feel we are going against the entity source pattern established by the previous two entity sources by requiring an instantiation for every time we want to resolve vs instantiating once and using it everywhere. If this is really something we don't mind and want to address as an optimization that's fine by me, just something I wanted to point out since it stood out to me.

Reply:
I'm a little confused as well.
- In our `main.go`, I'd expect to create the resolver with the "real" catalogd entity source, and then set the `reconciler.Resolver` in main.go
- In our tests, we're also seting up the reconciler, but there, we are using a mock (hardcoded) entity source. In that case, we setup the resolver with that entity source and set the `reconciler.Resolver` in the test file.

In both cases, the actual reconciler expects the resolver to be non-nil.